pkg/plugins/builtin/callback/handler_test.go

```diff
@@ -47,45 +71,64 @@ func TestMain(m *testing.M) {
 		panic(err)
 	}
 
+	if err := auth.Init(store); err != nil {
+		panic(err)
+	}
+
 	if err := compress.Register(); err != nil {
 		panic(err)
 	}
 
 	var wg sync.WaitGroup
+	ctx := context.Background()
 
-	if err := engine.Init(context.Background(), &wg, store); err != nil {
+	if err := engine.Init(ctx, &wg, store); err != nil {
 		panic(err)
 	}
 
-	step.RegisterRunner(Plugin.PluginName(), Plugin)
+	go srv.ListenAndServe()
+	srvx := &http.Server{Addr: fmt.Sprintf(":%d", utask.FPort)}
+	err := srvx.Shutdown(ctx)
+	if err != nil {
+		panic(err)
+	}
+
+	hdl = srv.Handler(ctx)
 
 	os.Exit(m.Run())
 }
 
-func TestHandleCallback(t *testing.T) {
-	dbp, err := zesty.NewDBProvider(utask.DBName)
-	assert.NoError(t, err)
+func dummyCustomMiddleware(c *gin.Context) {
+	c.Next()
+}
 
-	tt, err := tasktemplate.Create(dbp, "callback", "callback", nil, nil, []input.Input{}, []input.Input{}, []string{}, []string{}, false, false, nil, []values.Variable{}, nil, nil, "foo", nil, false, nil)
-	assert.NoError(t, err)
-	assert.NotNil(t, tt)
+func dumbIdentityProvider(r *http.Request) (string, error) {
+	username := r.Header.Get(usernameHeaderKey)
 
-	tsk, err := task.Create(dbp, tt, "foo", []string{}, []string{}, []string{}, []string{}, []string{}, nil, nil, nil)
-	assert.NoError(t, err)
-	assert.NotNil(t, t)
+	if username != adminUser && username != regularUser {
+		return "", errors.New("unknown user")
+	}
+	return r.Header.Get(usernameHeaderKey), nil
```

Comment:
Could just return `username` to spare an call.

Reply:
done!